Feat/accommodation_detail

Ryomi-j opened the pull request:
### PR 타입(하나 이상의 PR 타입을 선택해주세요)

- [x] 기능 추가
- [ ] 기능 삭제
- [x] 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 반영 브랜치

feature/accommodation_detail-> dev

### 변경 사항

- accommodationDetail component에서 Accommodation/Room info, Review section  컴포넌트 분리
- 숙소 상세, 비교모달 interface, typo 수정
-  room 가격이 0일 때 버튼 disabled
- 리뷰가 없을 때 보여줄, empty review box 추가
- review data 요청시, request start number 수정
- 상세 페이지에서 보여주는 숙소 이미지 갯수 수정 (5 -> 3)
- detail page url 수정
- accommodation detail data request uri 수정
- 상세파이지 리뷰에서 user email을 user nickname으로 대체

Ryomi-j committed:
:memo:  chore: add new addded properties to interfaces in accommodationDetail.ts

Ryomi-j committed:
:memo:  chore: fix typo

Ryomi-j committed:
:memo:  chore: fix typo -from totalElement to totalElements - and trim IReview propery order

Ryomi-j committed:
:memo:  chore: make IReviewResponseContentInitData

Ryomi-j committed:
:sparkles:  feat: make AccommodationInfo to use in  AccommodationDetail component

Ryomi-j committed:
:memo:  chore: make IRoomData interface

Ryomi-j committed:
:memo:  chore: make RoomInfo component

Ryomi-j committed:
:sparkles:  feat: make ReviewSection component

Ryomi-j committed:
:memo:  chore: add RoomInfo, ReviewSection and AccommodationInfo to AccommodationDetail component

Ryomi-j committed:
:memo:  chore: fix typo and erase unnecessary code

Ryomi-j committed:
:memo:  chore: add condition to render accommodation pictures

Ryomi-j committed:
:memo:  chore: fix response error - edit accommodationData data

Ryomi-j committed:
:memo:  chore: fix error - add IReviewResponseContentInitData to reviewRes.content and default value to reviewRes properties

Ryomi-j committed:
:memo:  chore: edit roomData default value and set room data after getting accomodation response

Ryomi-j committed:
:art:  style: change the number of accommodation images in grid frame

Ryomi-j committed:
:memo:  chore: add no-review box to ReviewSection component

Ryomi-j committed:
:memo:  chore: change the position of ReviewSection title tag

Ryomi-j committed:
:art:  style: apply style to table

Ryomi-j committed:
:memo:  chore: edit the review page start number when request data

Ryomi-j committed:
:art:  style: add cursor pointer style to summary tag

Ryomi-j committed:
:art:  style: apply prettier to AccommodationInfo component

Ryomi-j committed:
:art:  style: edit empty review box font size

Ryomi-j committed:
:memo:  chore: fix typo

Ryomi-j committed:
:memo:  chore: fix room image error - add condition to render room images

Ryomi-j committed:
:memo:  chore: make button disabled when room price is 0

Ryomi-j committed:
:memo:  chore: fix typo in ComparisonBox component

cloudedpanther reviewed (commented):
- 기본정보 부분도 ```split('</section>')```으로 끝내는게 좋을 것 같아요 ```'<!-- 편의시설 및 서비스 -->'```가 없으면 split시에 에러가 발생합니다. 또한 나머지 부분들도 split안의 string들이 존재하는지 검사하는 코드가 필요할 것 같습니다. 파악하지 못한 곳에서 에러가 날 수 있으니까요.
- 편의시설 부분도 section이 따로 있는 것으로 보입니다. ```'<section class="service">'```로 시작해서 ```'</section>'```으로 끝납니다.
- review page가 0부터 시작해야 합니다.

Ryomi-j committed:
:memo:  chore: fix key props error in RoomInfo component

Ryomi-j commented:
>  기본정보 부분도 split('</section>')으로 끝내는게 좋을 것 같아요 '<!-- 편의시설 및 서비스 -->'가 없으면 split시에 에러가 발생합니다.
: `<!-- 편의시설 및 서비스 -->`가 없어도 에러가 발생하지 않습니다.
: 'apple'.split('t') // ['apple']

>  또한 나머지 부분들도 split안의 string들이 존재하는지 검사하는 코드가 필요할 것 같습니다. 파악하지 못한 곳에서 에러가 날 수 있으니까요.
: 해당 부분은 api명세에 있는 내용과 response 여러개를 확인해 본 결과 공통으로 들어가는 부분을 사용했습니다.
: api명세에는 `[상세정보]
- 모텔 : defaultInfo(기본정보), feeInfo(요금정보), sellerInfo(판매자 정보)
- 호텔, 모텔 : defaultInfo(기본정보), serviceInfo(서비스 정보), sellerInfo(판매자 정보)`와 같이 나와있고, 해당 부분들은 공통된 주석으로 contents가 시작됩니다.
: 추가로 클롤링한 '여기어때'에서 개발자도구로 숙소정보 아코디언을 펼쳐 보시면, 공통된 주석으로 구분되어 있습니다.
: 여기어때 프론트 개발자들도 여러개의 데이터를 일반화하여 페이지를 개발을 했다고 생각해서 해당 부분이 따로 필요하다고 생각하지는 않습니다. 하지만 추가로 필요하다고 생각하시면 추가하겠습니다. 

> 편의시설 부분도 section이 따로 있는 것으로 보입니다. `<section class="service">`로 시작해서 `</section>`으로 끝납니다.
: 해당 부분은`<section class="service">`와 </section>로  split 하고 있습니다. 어떤 부분이 문제인지 다시 알려주실 수 있을까요? 참고로, section 태그로 편의시설 부분을 랩핑하지 않은 이유는 이미 div로 랩핑하고 있어 불필요한 랩핑을 방지하기 위함니다.


> review page가 0부터 시작해야 합니다.
: 해당 부분은 수정하겠습니다.

cloudedpanther commented:
> 기본정보 부분도 split('')으로 끝내는게 좋을 것 같아요 ''가 없으면 split시에 에러가 발생합니다.
> : `<!-- 편의시설 및 서비스 -->`가 없어도 에러가 발생하지 않습니다.
> : 'apple'.split('t') // ['apple']
* 'apple'.split('t')[1].split('p')의 경우를 말씀드린 것입니다. split이 연속으로 사용되는 경우가 있어서 우려했던 점입니다.
* api 명세대로 작성하셨다 하니 큰 상관은 없을 것 같습니다만 단지 방어코드적인 면에서 말씀드려봤습니다. 😀
* 나머지 부분도 괜찮을 것 같습니다! conflict만 잘 resolve 해주시면 될 것 같아요.

Ryomi-j commented:
>'apple'.split('t')[1].split('p')의 경우를 말씀드린 것입니다. split이 연속으로 사용되는 경우가 있어서 우려했던 점입니다.

: 해당 부분은 1번째 데이터가 없는 경우 split을 사용했기에 발생하는 오류 입니다. 
: 하지만 현재 사용하고 있는 데이터의 경우 데이터를 split한 후 필요한 데이터의 index와 해당 index의 데이터의 존재 유무도 명확하게 파악할 수 있다고 판단했습니다. 이유는 처음 데이터를 split을 한 후 1번 index의 데이터를 다듬는 과정이므로 항상 0번째 데이터는 존재하기 때문입니다.

Ryomi-j committed:
:memo:  chore: edit review page start number and related codes

Ryomi-j committed:
:memo:  chore: change how to show reviewer name - from email to nick name

Ryomi-j committed:
:memo:  chore: edit accommodation link url in SearchResult component

Ryomi-j committed:
:memo:  chore: use changed uri that is used when request accommodation detail data

Ryomi-j committed:
:memo:  chore: pull dev to resolve conflict

Ryomi-j committed:
:memo:  chore: add notice to let user know there is no room in the accommodation

Ryomi-j committed:
:art:  style: fix accommodation room image size error

Ryomi-j committed:
:art:  style: erase room title margin
